source/JsMaterialX/JsMaterialXGenGlsl/JsGlslShaderGenerator.cpp

```diff
@@ -13,7 +13,8 @@ namespace mx = MaterialX;
 
 EMSCRIPTEN_BINDINGS(GlslShaderGenerator)
 {
-    ems::class_<mx::GlslShaderGenerator, ems::base<mx::ShaderGenerator>>("GlslShaderGenerator")
-        .smart_ptr_constructor("GlslShaderGenerator", &std::make_shared<mx::GlslShaderGenerator>)
+    ems::class_<mx::GlslShaderGenerator, ems::base<mx::HwShaderGenerator>>("GlslShaderGenerator")
+        .constructor<mx::TypeSystemPtr>()
+        .class_function("create", &mx::GlslShaderGenerator::create)
```

Comment:
I can see that it's tricky to provide JavaScript bindings for these `create` functions, in part because their default arguments are _function calls_ rather than constant values.

As an alternative, what if the default arguments to these `create` functions were simply `nullptr`, and the create functions would internally call `TypeSystem::create` when they saw a null `typeSystem` input?

This would make it more straightforward to provide JavaScript (and Python) bindings for these create methods, and would allow us to simplify the new logic in our JavaScript test suite above.

Reply:
That's a good idea! I was struggling to make the JavaScript bindings working again. But this should simplify things.